s3/src/main/scala/monix/connect/s3/S3.scala

```diff
@@ -885,6 +884,111 @@ trait S3 { self =>
+    * @param prefix        limits the response to keys that begin with the specified prefix.
+    * @param requestPayer  confirms that the requester knows that she or he will be charged for
+    *                      the list objects request in V2 style.
+    *                      Bucket owners need not specify this parameter in their requests.
+    * @return an [[Observable]] that emits the [[S3Object]]s.
+    */
+
+  def listLatestObject(
+    bucket: String,
+    prefix: Option[String] = None,
+    requestPayer: Option[RequestPayer] = None): Observable[S3Object] = {
+    val sort: (S3Object, S3Object) => S3Object = (a, b) =>
+      if (a.lastModified().compareTo(b.lastModified()) >= 0) a else b
+    for {
+      listResponse <- ListObjectsObservable(bucket, prefix, None, requestPayer, this.s3Client)
+      s3Object     <- Observable.fromIterable(listResponse.contents.asScala).reduce(sort)
+    } yield s3Object
+  }
+
+  /** Returns oldest N objects in bucket.
+    *
+    * ==Example==
+    *
+    * {{{
+    * import cats.effect.Resource
+    * import monix.connect.s3.S3
+    * import monix.eval.Task
+    * import software.amazon.awssdk.auth.credentials.{AwsBasicCredentials, StaticCredentialsProvider}
+    * import software.amazon.awssdk.regions.Region
+    * import monix.execution.Scheduler.Implicits.global
+    *
+    * val bucket = "my-bucket"
+    * val prefix = "prefix/to/list/keys"
+    * val s3AccessKey: String = "TESTKEY"
+    * val s3SecretKey: String = "TESTSECRET"
+    * val basicAWSCredentials = AwsBasicCredentials.create(s3AccessKey, s3SecretKey)
+    * val staticCredProvider = StaticCredentialsProvider.create(basicAWSCredentials)
+    *
+    * val s3: Resource[Task, S3] = S3.create(staticCredProvider, Region.AWS_GLOBAL)
+    *
+    * s3.use{s3 => s3.listOldestNObjects(bucket, 6, prefix = Some(prefix)).toListL}.runToFuture
+    * }}}
+    *
+    * To use this operation in an AWS (IAM) policy, you must have permissions to perform
+    * the `ListBucket` action. The bucket owner has this permission by default and can grant it.
+    *
+    * @param bucket        target S3 bucket name of the object to be downloaded.
+    * @param amount        the number of objects you would like to return
+    * @param prefix        limits the response to keys that begin with the specified prefix.
+    * @param requestPayer  confirms that the requester knows that she or he will be charged for
+    *                      the list objects request in V2 style.
+    *                      Bucket owners need not specify this parameter in their requests.
+    * @return an [[Observable]] that emits the [[S3Object]]s.
+    */
+
+  def listOldestNObjects(
+    bucket: String,
+    n: Int,
+    prefix: Option[String] = None,
+    requestPayer: Option[RequestPayer] = None): Observable[S3Object] = {
+    val sorted: (S3Object, S3Object) => Boolean = (x, y) =>
+      if (x.lastModified().compareTo(y.lastModified()) < 0) true else false
```

Comment:
maybe just below code is enough:
```
val sorted: (S3Object, S3Object) => Boolean = (x, y) => (x.lastModified() < y.lastModified())
```

Reply:
< is not a member of java.time.Instant. So I would have to do:
```
(x.lastModified().toEpochMilli() < y.lastModified().toEpochMill()
```
which is cool, whatever you prefer.

Reply:
oh yeah, then we could just do:
```
(x.lastModified().compareTo(y.lastModified())
```

Reply:
I think would be good to address this one, makes it more readable to me :)

Reply:
```(x.lastModified().compareTo(y.lastModified())``` is going to spit out 0, 1, or some negative int that's why I've kept it this way.

Reply:
but i did remove the the if statement